### ISSUE TITLE
False negative: Nested blocks inside an if/else clause are not detected

### ISSUE THREAD

**Issue by oeken** [rrrene/credo#1029]
### Environment

Credo version: `1.6.7`
Erlang/Elixir version: 
```
Erlang/OTP 25 [erts-13.1.5] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit] [dtrace]
Elixir 1.14.3 (compiled with Erlang/OTP 25)
```
Operating system: macos M1

### What were you trying to do?

Identify deeply nested functions. I used `mix credo --all` with default configuration. It successfully identified many. However missed functions of the following form. 

```
defp foo() do
  if true do
    if true do
      if true do
        if true do
          "nested"
        end
      end
    end
  else
    "flat"
  end
end
```

Note: If I omit the `else` block then the suggestion is raised.

### Expected outcome

Raise suggestion: `Function body is nested too deep (max depth is 2, was 4).`

### Actual outcome

Credo found no issues.

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by oeken** [comment]
Yes! Thank you for speedy fix 🚀

**Comment by oeken** [comment]
Reopening because the code change indeed addresses nesting inside `if` blocks but not inside `else` blocks. e.g. 
```
defp foo() do
  if true do
    "flat"
  else
    if true do
      if true do
        if true do
          "nested"
        end
      end
    end
  end
end
```
could you take a look? 👀 thank you 🙏🏻

**Comment by rrrene** [comment]
D'oh! Just pushed another fix. Could you try again?
